### PR TITLE
Mutual info render rework, copy statusbar text

### PR DESCRIPTION
**Pull request by aronhelser**
@vibraphone @jourdain 
If you have time, I'd appreciate an independent test. I made the MI diagram do an update during the render() call, instead of a complete re-build of the svg element. I think I got everything.... :)

Also added the click-dblclick chord fix and automatically selecting the status bar text on click so it can be copied, if desired.

### CODE REVIEW

File: src/InfoViz/Native/MutualInformationDiagram/index.js [hunk 1]
```diff
@@ -46,8 +48,8 @@ function informationDiagram(publicAPI, model) {
 
   model.clientRect = null;
 
-  // FIXME: Make some attempt at unique id, for now just use millis timestamp
-  model.instanceID = `informationDiagram-${Date.now()}`;
+  // unique id, based on millis timestamp
+  model.instanceID = `informationDiagram-${generateUUID()}`;
```
Comment (jourdain) [inline src/InfoViz/Native/MutualInformationDiagram/index.js]: Why not just a global counter within the module? No complexity and no risk of same id.

**Comment by jourdain** [comment]
I let David test it out otherwise code wise, it seems fine to me.

File: src/InfoViz/Native/MutualInformationDiagram/index.js (hunk 1 above)
Comment (aronhelser) [inline src/InfoViz/Native/MutualInformationDiagram/index.js]: OK, seems fine. It's used for an xlink:href target ID, so it has to be unique in the entire page, but a count with the module string should be fine.

File: src/InfoViz/Native/MutualInformationDiagram/index.js [hunk 2]
```diff
@@ -139,7 +141,23 @@ function informationDiagram(publicAPI, model) {
...
+  publicAPI.selectStatusBarText = () => {
+    // select text so user can press ctrl-c if desired.
+    if (model.statusBarVisible) {
+      // https://www.sitepoint.com/javascript-copy-to-clipboard/
+      const range = document.createRange();
+      range.selectNode(d3.select(model.container).select('span.status-bar-text').node());
+      window.getSelection().removeAllRanges();
+      window.getSelection().addRange(range);
+      // Copy-to-clipboard doesn't work unless status bar text is an 'input':
+      // document.execCommand('copy');
```
Comment (aronhelser) [inline src/InfoViz/Native/MutualInformationDiagram/index.js]: I tried copy-paste into input widgets on Edge, Chrome and Firefox, and they all seem to handle the special characters. So might be worth trying. Do you think destroying the user's clipboard is acceptable in this case?

Comment (jourdain) [inline src/InfoViz/Native/MutualInformationDiagram/index.js]: I think so

**Comment by aronhelser** [comment]
OK, updated to use a re-style read-only input element in the status bar, and copy-to-clipboard works well. 

I'll wait to merge until David has a chance to try it out, though.

**Comment by vibraphone** [comment]
+2

**Pull request merged**